### ISSUE TITLE
Basic auth web in prometheus and alertmanager

### ISSUE THREAD

**Issue by rajibmitr** [prometheus-operator/prometheus-operator#5744]
So we have been working with later version of prometheus-operator and found out below 

 
kubectl explain alertmanager.spec.alertmanagerConfiguration.global.httpConfig.basicAuth.password
kubectl explain alertmanager.spec.alertmanagerConfiguration.global.httpConfig.basicAuth.username

can anyone please confirm are these same as basic auth web which is mentioned here - https://prometheus.io/docs/guides/basic-auth/

Or these are totally different ?

**Comment by slashpai** [comment]
Could you please tell what you are trying to do?

**Comment by rajibmitr** [comment]
I see this PR is still not merged - https://github.com/prometheus-operator/prometheus-operator/pull/4942

**Comment by slashpai** [comment]
> So we have been working with later version of prometheus-operator and found out below
> 
> kubectl explain alertmanager.spec.alertmanagerConfiguration.global.httpConfig.basicAuth.password kubectl explain alertmanager.spec.alertmanagerConfiguration.global.httpConfig.basicAuth.username
> 
> can anyone please confirm are these same as basic auth web which is mentioned here - https://prometheus.io/docs/guides/basic-auth/
> 
> Or these are totally different ?

This basicAuth is for https://prometheus.io/docs/alerting/latest/configuration/#http_config

**Comment by rajibmitra** [comment]
> 

And what is this alertmanager.spec.alertmanagerConfiguration.global.httpConfig.basicAuth ? @slashpai

**Comment by rajibmitra** [comment]
I am little confused, https://github.com/prometheus-operator/prometheus-operator/pull/4942 - once this PR is merged then we can enable the basic auth - correct @slashpai  ?

**Comment by slashpai** [comment]
> > 
> 
> And what is this alertmanager.spec.alertmanagerConfiguration.global.httpConfig.basicAuth ? @slashpai

Same as answered above, AlertmanagerGlobalConfig configures parameters that are valid in all other configuration contexts. see [here](https://prometheus.io/docs/alerting/latest/configuration/#configuration-file)


https://github.com/prometheus-operator/prometheus-operator/blob/release-0.66/pkg/apis/monitoring/v1/alertmanager_types.go#L279 -> https://github.com/prometheus-operator/prometheus-operator/blob/release-0.66/pkg/apis/monitoring/v1/alertmanager_types.go#L356

**Comment by rajibmitra** [comment]
thanks @slashpai https://github.com/prometheus-operator/prometheus-operator/blob/release-0.66/pkg/apis/monitoring/v1/alertmanager_types.go#L349 - this says its for the client, actually I want to enable basic auth for self when we try to open alertmanager thru' browser.

**Comment by rajibmitra** [comment]
@slashpai this is what I am trying to do - https://prometheus.io/docs/alerting/latest/https/#https-and-authentication

**Comment by slashpai** [comment]
Currently TLS and HTTPConf is supported https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#webconfigfilefields. 

https://github.com/prometheus-operator/prometheus-operator/pull/4942 was to do this for basic_auth_users

**Comment by rajibmitra** [comment]
thanks @slashpai !

**Comment by slashpai** [comment]
Closing this in favour of https://github.com/prometheus-operator/prometheus-operator/issues/4200
